src/Avalonia.Visuals/Rendering/DirtyVisuals.cs

```diff
@@ -60,6 +65,12 @@ public void Add(IVisual visual)
         /// </summary>
         public void Clear()
         {
+            if (_deferring)
+            {
+                _deferredChanges.Add((DeferredChange.Clear, null));
```

Comment:
The only time that the deferred changes are cleared is after the client has finished enumerating them. I don't think it know if it makes sense to add the `Clear` as a deferred change here. Probably clearing while iterating should be an error.